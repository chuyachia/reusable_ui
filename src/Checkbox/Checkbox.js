import React from "react";
import PropTypes from "prop-types";

const Checkbox = ({
  label,
  value,
  className,
  onChange,
  selected,
  groupSelection,
}) => {
  var props = { value, onChange };
  if (selected || groupSelection.indexOf(value) >= 0) props.checked = true;
  return (
    <label className={className}>
      {label}&nbsp;
      <input type="checkbox" {...props} />
      <span />
    </label>
  );
};

Checkbox.propTypes = {
  label: PropTypes.string,
  className: PropTypes.string,
  value: PropTypes.string,
  selected: PropTypes.bool,
  groupSelection: PropTypes.arrayOf(PropTypes.string),
  onChange: PropTypes.func,
};

export default Checkbox;
