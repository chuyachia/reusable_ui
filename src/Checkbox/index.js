import styled from "@emotion/styled";

import useTheme from "../useTheme";
import Checkbox from "./Checkbox";

export default styled(Checkbox)`
  display: ${props =>
    props.direction == "horizontal" ? "inline-block" : "block"};

  position: relative;
  cursor: pointer;
  font-family: ${props => useTheme(props, "fontSans")};
  padding-left: 1.5em;
  & input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
  }
  & span {
    position: absolute;
    top: 0;
    left: 0;
    height: 1em;
    width: 1em;
    background-color: ${props => useTheme(props, "silver")};
    border-radius: ${props => useTheme(props, "baseRadius")};
  }
  &:hover span {
    background-color: ${props => {
      switch (props.variant) {
        case "secondary":
          return useTheme(props, "secondaryLight");
        case "warning":
          return useTheme(props, "warningLight");
        default:
          return useTheme(props, "primaryLight");
      }
    }};
  }
  & input:checked ~ span {
    background-color: ${props => {
      switch (props.variant) {
        case "secondary":
          return useTheme(props, "secondary");
        case "warning":
          return useTheme(props, "warning");
        default:
          return useTheme(props, "primary");
      }
    }};
  }
  span:after {
    content: "";
    position: absolute;
    display: none;
  }
  & input:checked ~ span:after {
    display: block;
  }
  & span:after {
    top: 50%;
    left: 50%;
    width: 0.3em;
    height: 0.5em;
    border: solid ${props => useTheme(props, "silver")};
    border-width: 0 0.2rem 0.2rem 0;
    transform: translate(-50%, -75%) rotate(45deg);
  }
`;
