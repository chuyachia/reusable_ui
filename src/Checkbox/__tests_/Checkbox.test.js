import React from "react";

import Checkbox from "../Checkbox";

describe("Checkbox", () => {
  let wrapper;
  it("should render", () => {
    wrapper = shallow(<Checkbox />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render its children with properties", () => {
    var testclassname = "testclassname";
    var testvalue = "testvalue";
    wrapper = shallow(<Checkbox className={testclassname} value={testvalue} />);
    expect(wrapper.find("label").prop("className")).toBe(testclassname);
    expect(wrapper.find("input").prop("value")).toBe(testvalue);
  });
  it("should render a checked checkbox if prop selected is true", () => {
    wrapper = shallow(<Checkbox selected={true} />);
    expect(wrapper.find("input").prop("checked")).toBe(true);
  });
  it("should render a checked checkbox if its value is in prop groupSelection", () => {
    var testvalue = "testvalue";
    var groupSelection = ["testvalue", "aaa"];
    wrapper = shallow(
      <Checkbox value={testvalue} groupSelection={groupSelection} />
    );
    expect(wrapper.find("input").prop("checked")).toBe(true);
  });
});
