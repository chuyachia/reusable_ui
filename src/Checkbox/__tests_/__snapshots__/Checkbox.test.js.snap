// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render 1`] = `
<label>
  Â 
  <input
    onChange={[Function]}
    type="checkbox"
  />
  <span />
</label>
`;
